python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -333,11 +334,58 @@ class AssetArgs(NamedTuple):
     owners: Optional[Sequence[str]]
 
 
+@dataclass
+class ResourceRelatedState:
+    asset_args: AssetArgs
```

Comment:
Thoughts on narrowing this to just the resource-related properties – `io_manager_def`, `io_manager_key`, etc? That would help make it much more clear what state is actually "resource-related".